### PR TITLE
MGMT-9122: remove v1 EnableHost DisableHost

### PR DESCRIPTION
**Pull request by slaviered**
EnableHost/DisableHost are APIs that were replaced in V2
by either unbinding to deregistration.

This PR returns 404 when those APIs are called and removes
unnecessary code branches that implemented the old APIs.

Tests that verify those APIs were removed.
In other cases, EnableHost or DisableHost were used to induce
an error or check route processing.
Then, they were replaced with other calls.

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by filanov** [comment]
do a grep on HostStatusDisabled and make sure that it doesn't exist anymore it's not relevant for v2

**Comment by filanov** [comment]
LGTM, @nmagnezi i prefer that you would give another look

File: pkg/auth/authz_handler_test.go [hunk 1]
```diff
@@ -821,16 +816,6 @@ func enableHost(ctx context.Context, cli *client.AssistedInstall) error {
 	return err
 }
 
-func disableHost(ctx context.Context, cli *client.AssistedInstall) error {
```
Comment (nmagnezi) [inline pkg/auth/authz_handler_test.go:824]: what about `enableHost()` in line 814 at this file?

File: pkg/auth/authz_handler_test.go [hunk 2]
```diff
@@ -404,11 +404,6 @@ var _ = Describe("authz", func() {
 			allowedRoles: []ocm.RoleType{ocm.AdminRole, ocm.UserRole},
 			apiCall:      enableHost,
 		},
-		{
```
Comment (nmagnezi) [inline pkg/auth/authz_handler_test.go:407]: the test involving `enableHost()` in line 405 should probably go away as well

File: internal/ignition/ignition.go [hunk 3]
```diff
@@ -1039,8 +1039,6 @@ func sortHosts(hosts []*models.Host) ([]*models.Host, []*models.Host) {
 	workers := []*models.Host{}
 	for i := range hosts {
 		switch {
```
Comment (nmagnezi) [inline internal/ignition/ignition.go:1041]: Given that now we have just a case and a default, does switch-case still makes sense here?

**Review by nmagnezi — CHANGES_REQUESTED** [review]

**Review by nmagnezi — APPROVED** [review]
/lgtm

**Pull request merged**